### PR TITLE
[BE] Code motion - tracing, compile context, etc, to their own file

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114317

TLS

Lint

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

File: torch/_dynamo/variables/nn_module.py [hunk 1]
```diff
@@ -74,7 +74,12 @@ def record_nn_module_stack(module_key: str, source, tx, mod: torch.nn.Module):
 
 
 class NNModuleVariable(VariableTracker):
-    _nonvar_fields = {"module_type", "module_key", "module", *VariableTracker._nonvar_fields}
+    _nonvar_fields = {
+        "module_type",
+        "module_key",
+        "module",
+        *VariableTracker._nonvar_fields,
+    }
```
Comment (ezyang) [inline torch/_dynamo/variables/nn_module.py:82]: please don't reformat completely unrelated files on refactors like this, stuff like this is why I have to carefully review BE PRs from you lol

**Review by ezyang — APPROVED** [review]
ok

**Comment by voznesenskym** [comment]
Cant merge because of a weird executorch dep that is using a private API, and is being used as an external package? Going to ping @mergennachin and @tugsbayasgalan about this...
